### ISSUE TITLE
testsupport: allow passing additional arguments when launching the debug adapter process

### ISSUE THREAD

**Issue by justarandomgeek** [microsoft/vscode-debugadapter-node#285]
My debug adapter is part of a multi-function tool, so it needs additional command line arguments to start in DAP mode, but testsupport's `DebugClient` doesn't appear to have anywhere to pass these additional arguments.

**Comment by roblourens** [comment]
Isn't this what `spawnOptions` is? If not can you be more specific?

**Comment by justarandomgeek** [comment]
no, spawnOptions is for all the *other* properties to pass to spawn, but ther's no `argv` in it, which is what i need to extend. 

For now i've done the hideous hack of passing an array of extra args through an env var: https://github.com/justarandomgeek/vscode-factoriomod-debug/blob/next/test/debug.ts#L15
https://github.com/justarandomgeek/vscode-factoriomod-debug/blob/next/src/cli/main.ts#L22

**Comment by justarandomgeek** [comment]
https://github.com/microsoft/vscode-debugadapter-node/blob/main/testSupport/src/debugClient.ts#L97 builds the argv array from only `debugAdapterExecutable`, leaving no opportunity to extend it

**Comment by roblourens** [comment]
Oh, got it. Makes perfect sense, I'd take a PR that adds this. Would have to be another optional argument to the end of the list of constructor args.

**Comment by roblourens** [comment]
I'll fix it if I don't get a PR. It would just be faster if someone sent a PR.

**Comment by justarandomgeek** [comment]
any chance of this get fixed any time soon? Would be nice to get rid of my weird hacks around it...
